### PR TITLE
CB-15857 Add EDH feature tracking capability to FreeIPA service and t…

### PR DESCRIPTION
**Pull request by nadamdb**
…rack FreeIPA up and downscaling actions.

Previously the EDH feature tracking was only implemented in the core and environment services. With this change the tracking also becomes available in the freeipa service. The implemetation is based on the existing implementations in the other two services. The flow state mapping is similar to the mapping in the legacy structuredevent service, because the capability to differentiate single flows from flows in a flowchain is necessary for the distinction of a standalone freeipa scaling action from scaling during a freeipa upgrade (tracking will be added later).

See detailed description in the commit message.

### CODE REVIEW

**Comment by lnardai** [comment]
Is this essential for 2.54? Looks like a new feature to me. If so, please retarged this to master.

**Comment by keyki** [comment]
> Is this essential for 2.54? Looks like a new feature to me. If so, please retarged this to master.

It's not essential to 2.54, we will need it to track the feature in EDH. We can do this as a hotfix, don't block the release because of this, but we eventually need this in 2.54.

**Comment by lnardai** [comment]
@keyki should we wait for a review from your team?

**Comment by keyki** [comment]
> @keyki should we wait for a review from your team?

Yes, @pdarvasi will review today

File: usage-collection/src/main/java/com/sequenceiq/cloudbreak/usage/UsageReporter.java
Comment (pdarvasi) [inline usage-collection/src/main/java/com/sequenceiq/cloudbreak/usage/UsageReporter.java:70]: minor typo: `an` -> `a`. Pls replace all occurrences

**Review by pdarvasi — CHANGES_REQUESTED** [review]

File: structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/converter/CDPEnvironmentStructuredFlowEventToCDPEnvironmentRequestedConverter.java
Comment (nadamdb) [reply structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/converter/CDPEnvironmentStructuredFlowEventToCDPEnvironmentRequestedConverter.java:36]: The payload in CDPStructuredEvent class is generic type and operationDetailsConverter.convert() converts only the CDPStructuredEvent class. The payload in the converter previously was only used to get the cloud provider, but before the change the converter was not converting CDPStructuredEvent but the child class CDPEnvironmentStructuredFlowEvent so the payload type was known. By making the converter more generic the type is not known anymore. I tried making a separate abstract Payload class that has a getCloudProvider() function, but in the datalake service SdxClusterDto (in datalake module) is used as payload and making that class extend the Payload class (in structuredeventmodel module) was causing cyclic dependency. 
Addig it as parameter to the converter was the next best thing that i could think of, but i'm open to suggestions for a better solution.

**Review by pdarvasi — APPROVED** [review]

**Comment by pdarvasi** [comment]
LGTM, nicely done, @nadamdb

**Review by keyki — APPROVED** [review]

**Review by keyki — APPROVED** [review]

**Pull request merged**
